src/main/java/teammates/common/datatransfer/attributes/AccountAttributes.java

```diff
@@ -26,49 +27,109 @@
     public Date createdAt;
     public StudentProfileAttributes studentProfile;
 
-    public AccountAttributes(Account a) {
-        googleId = a.getGoogleId();
-        name = a.getName();
-        isInstructor = a.isInstructor();
-        email = a.getEmail();
-        institute = a.getInstitute();
-        createdAt = a.getCreatedAt();
-        studentProfile =
-                a.getStudentProfile() == null ? null : StudentProfileAttributes.valueOf(a.getStudentProfile());
-    }
-
     public AccountAttributes() {
         // attributes to be set after construction
     }
 
-    public AccountAttributes(String googleId, String name, boolean isInstructor,
-                String email, String institute, StudentProfileAttributes studentProfileAttributes) {
+    public AccountAttributes(String googleId, String name, boolean isInstructor, String email, String institute) {
         this.googleId = SanitizationHelper.sanitizeGoogleId(googleId);
         this.name = SanitizationHelper.sanitizeName(name);
         this.isInstructor = isInstructor;
         this.email = SanitizationHelper.sanitizeEmail(email);
         this.institute = SanitizationHelper.sanitizeTitle(institute);
-        this.studentProfile = studentProfileAttributes;
-        this.studentProfile.sanitizeForSaving();
+    }
 
+    public static Builder builder(String googleId, String name, boolean isInstructor, String email, String institute) {
+        return new Builder(googleId, name, isInstructor, email, institute);
     }
 
-    public AccountAttributes(String googleId, String name, boolean isInstructor,
-                String email, String institute) {
-        this.googleId = SanitizationHelper.sanitizeGoogleId(googleId);
-        this.name = SanitizationHelper.sanitizeName(name);
-        this.isInstructor = isInstructor;
-        this.email = SanitizationHelper.sanitizeEmail(email);
-        this.institute = SanitizationHelper.sanitizeTitle(institute);
-        this.studentProfile = StudentProfileAttributes.builder().build();
-        this.studentProfile.googleId = this.googleId;
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private static final String REQUIRED_FIELD_CANNOT_BE_NULL = "Non-null value expected";
+        private AccountAttributes accountAttributes;
+
+        public Builder(String googleId, String name, boolean isInstructor, String email, String institute) {
+            validateRequiredFields(googleId, name, isInstructor, email, institute);
+            accountAttributes = new AccountAttributes(googleId, name, isInstructor, email, institute);
+        }
+
+        public Builder() {
+            accountAttributes = new AccountAttributes();
+        }
+
+        public AccountAttributes build() {
+            return accountAttributes;
```

Comment:
I just realized that there is no "compulsory" feild because there is no parameter which is always !null, so I think I will get rid of `validateRequiredFields`